### PR TITLE
Feature/phase dates rework

### PR DESCRIPTION
**Pull request by JakeNunemaker**

### CODE REVIEW

**Comment by JakeNunemaker** [comment]
- Fixed bug with `ProjectManager.phase_dates` when using date string and dependent phases.
- Added check for mixed input types in dependent phases
- Added ability to define fixed offsets in dependent phases, eg. `hours=10` or `days=1;hours=10`
- Added tests for fixed offsets
- Added tests for `ProjectManager.phase_dates`

File: ORBIT/manager.py [hunk 1]
```diff
@@ -1141,7 +1194,12 @@ def phase_dates(self):
 
         for phase, _start in self.config["install_phases"].items():
 
-            start = dt.datetime.strptime(_start, self.date_format_short)
+            try:
+                start = dt.datetime.strptime(_start, self.date_format_short)
+
+            except TypeError:
+                start = self.start_date + dt.timedelta(hours=ceil(self.phase_starts[phase]))
```
Comment (ashesh2512) [inline ORBIT/manager.py:1201]: Could we move away from `ceil` for both `start` and `end` ? It won't result in well rounded off times, but should match the action logs exactly.

**Review by ashesh2512 — APPROVED** [review]
This looks good to me!

File: ORBIT/manager.py (hunk 1 above)
Comment (JakeNunemaker) [reply ORBIT/manager.py]: I removed it here and in a few different places and ran the tests - everything still works. I had to keep it in the weather profile indexing but that shouldn't affect these outputs. It just internally rounds the time up to the nearest index location when getting a weather profile for an install phase.

**Review by ashesh2512 — APPROVED** [review]

**Pull request merged**